### PR TITLE
docs: improvements to ICA docs and v6 migration

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

A series of improvements to v6 migration and ICA docs with the goal of making it more explicit about the migration paths and the recommended ways of using ICA depending on the use case.

[Rendered view](https://github.com/cosmos/ibc-go/blob/carlos/ica-docs-v6-migration-improvements/docs/migrations/v5-to-v6.md) of the migration docs.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
docs: improvements to ICA docs and v6 migration
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -234,19 +244,36 @@ module.exports = {
...
                 {
                   title: "Active Channels",
                   directory: false,
                   path: "/apps/interchain-accounts/active-channels.html",
                 },
                 {
-                  title: "Authentication Modules",
-                  directory: false,
-                  path: "/apps/interchain-accounts/auth-modules.html",
+                  title: "Legacy",
```
Comment (crodriguezvega) [inline docs/.vuepress/config.js:257]: I added like a subfolder with the legacy related documentation, so that when we are ready to eliminate the legacy APIs, we can just remove these files at that should be it (from a documentation updating point of view).

File: docs/apps/interchain-accounts/client.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
+<!--
+order: 7
+-->
+
+# Client
```
Comment (crodriguezvega) [inline docs/apps/interchain-accounts/client.md:5]: I added this file for all CLI and grpc queries following the example of the SDK (e.g. for [`x/bank`](https://github.com/cosmos/cosmos-sdk/tree/main/x/bank#client)).

File: docs/apps/interchain-accounts/overview.md [hunk 3]
```diff
@@ -20,16 +20,14 @@ Regular accounts use a private key to sign transactions. Interchain Accounts are
...
+`Interchain Account`: An account on a host chain created using the ICS27 protocol. An interchain account has all the capabilities of a normal account. However, rather than signing transactions with a private key, a controller chain will send IBC packets to the host chain which signals what transactions the interchain account should execute.
 
-#### Deprecated
+`Authentication Module`: A custom application module on the controller chain that uses the Interchain Accounts module to build custom logic for the creation & management of interchain accounts. It can be either an IBC application module using the [legacy API](./legacy/keeper-api.md), or a regular Cosmos SDK application module sending messages to the controller submodule's `MsgServer` (this is the recommended approach from ibc-go v6 if access to packet callbacks is not needed).
 
-`Authentication Module`: A custom IBC application module on the controller chain that uses the Interchain Accounts module API to build custom logic for the creation & management of interchain accounts. For a controller chain to utilize the interchain accounts module functionality, an authentication module is required.
+## SDK security model
```
Comment (colin-axner) [inline docs/apps/interchain-accounts/overview.md:27]: I think modifying the titles or changing `ICS-27` to `ICS27` are example of formatting changes that could go in separate prs

File: docs/apps/interchain-accounts/client.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+## CLI
```
Comment (colin-axner) [inline docs/apps/interchain-accounts/client.md:7]: Moving the CLI documentation and/or adding documentation to it (unsure if that happened) in the future should be done in a separate pr to main. This goes for any logically isolated concept

File: docs/apps/interchain-accounts/integration.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <!--
-order: 2
+order: 4
 -->
 
 # Integration
```
Comment (colin-axner) [inline docs/apps/interchain-accounts/integration.md:5]: This could be a separate pr to main (changes to app.go)

File: docs/apps/interchain-accounts/active-channels.md [hunk 6]
```diff
@@ -1,29 +1,33 @@
...
 
-It is possible to create a new channel using the same controller chain portID if the previously set `Active Channel` is now in a `CLOSED` state. This channel creation can be initialized programatically by sending a new `MsgChannelOpenInit` message like so:
+When an Interchain Account is registered using `MsgRegisterInterchainAccount`, a new channel is created on a particular port. During the `OnChanOpenAck` and `OnChanOpenConfirm` steps (on controller & host chain respectively) the `Active Channel` for this interchain account is stored in state.
+
+It is possible to create a new channel using the same controller chain portID if the previously set `Active Channel` is now in a `CLOSED` state. This channel creation can be initialized programatically by sending a new `MsgRegisterInterchainAccount` message like so:
 
 ```go
-msg := channeltypes.NewMsgChannelOpenInit(portID, string(versionBytes), channeltypes.ORDERED, []string{connectionID}, icatypes.PortID, icatypes.ModuleName)
-handler := k.msgRouter.Handler(msg)
+msg := controllertypes.NewMsgRegisterInterchainAccount(connectionID, controllerAddress, version)
```
Comment (colin-axner) [inline docs/apps/interchain-accounts/active-channels.md:16]: Hmm, I don't think this is true. This should fail if the account was created with the legacy API. I would revert these changes

**Review by colin-axner — COMMENTED** [review]
Thanks @crodriguezvega! I only had the mental energy to review 6/15 files. I think it would be very beneficial for future reviewers if you could pull out some of these changes. I think even opening a pr for each file when possible could help a lot. I think you could separate ordering changes from documentation improvements. If you pull out changes, I think I'd prefer if they are all based on main. I think it is fine to have unfinished changes merged into main (it is our development branch)

File: docs/apps/interchain-accounts/parameters.md [hunk 7]
```diff
@@ -1,53 +1,54 @@
...
 
 | Key                    | Type | Default Value |
 |------------------------|------|---------------|
 | `ControllerEnabled`    | bool | `true`        |
 
-#### ControllerEnabled
+### ControllerEnabled
 
-The `ControllerEnabled` parameter controls a chains ability to service ICS-27 controller specific logic. This includes the sending of Interchain Accounts packet data as well as the following ICS-26 callback handlers:
+The `ControllerEnabled` parameter controls a chains ability to service ICS27 controller specific logic. This includes the sending of Interchain Accounts packet data as well as the following ICS-26 callback handlers:
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/parameters.md:17]: I was trying to find a consistent way to write it to because in same places it's written as `ICS-27` in other places `ICS27` and yet in other places as `ICS 27`. I saw your comment of preferring `ICS 27` or `ICS-27` and I am happy with both, so I will change it back.

File: docs/apps/interchain-accounts/parameters.md [hunk 8]
```diff
@@ -1,53 +1,54 @@
...
 - `OnRecvPacket`
 
-#### AllowMessages
+### AllowMessages
 
-The `AllowMessages` parameter provides the ability for a chain to limit the types of messages or transactions that hosted interchain accounts are authorized to execute by defining an allowlist using the Protobuf message TypeURL format.
+The `AllowMessages` parameter provides the ability for a chain to limit the types of messages or transactions that hosted interchain accounts are authorized to execute by defining an allowlist using the Protobuf message type URL format.
 
-For example, a Cosmos SDK based chain that elects to provide hosted Interchain Accounts with the ability of governance voting and staking delegations will define its parameters as follows:
+For example, a Cosmos SDK-based chain that elects to provide hosted Interchain Accounts with the ability of governance voting and staking delegations will define its parameters as follows:
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/parameters.md:43]: I will remove it.

File: docs/apps/interchain-accounts/active-channels.md (hunk 6 above)
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/active-channels.md:16]: All right, thanks. So the recommended way for ibc-go >= v6 is then to use `NewMsgChannelOpenInit`, although for ibc-go < v6 is was possible to use `NewMsgRegisterInterchainAccount`, right?

File: docs/migrations/v5-to-v6.md [hunk 9]
```diff
@@ -87,21 +87,86 @@ app.UpgradeKeeper.SetUpgradeHandler(
...
 
-Legacy APIs are still required if application developers wish to consume IBC packet callbacks and react upon packet acknowledgements. In future this will be replaced by IBC Actor Callbacks, see [ADR 008](https://github.com/cosmos/ibc-go/pull/1976).
+Both APIs remain functional and maintain backwards compatibility in ibc-go v6, however consumers of these APIs are now recommended to follow the message passing paradigm outlined in Cosmos SDK [ADR 031](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-031-msg-service.md) and [ADR 033](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-033-protobuf-inter-module-comm.md). This is facilitated by the Cosmos SDK [`MsgServiceRouter`](https://github.com/cosmos/cosmos-sdk/blob/main/baseapp/msg_service_router.go#L17) and chain developers creating custom application logic can now omit the ICS27 controller submodule `Keeper` from their module and instead depend on message routing. 
 
-For more information see the new [ICS27 integration documentation](TODO: add link to new docs).
+Depending on the use case, developers of custom authentication modules have then three possibilities:
```
Comment (tmsdkeys) [inline docs/migrations/v5-to-v6.md:103]:
```diff
- Depending on the use case, developers of custom authentication modules have then three possibilities:
+ Depending on the use case, developers of custom authentication modules face one of three scenarios:
```

File: docs/migrations/v5-to-v6.md [hunk 10]
```diff
@@ -87,21 +87,86 @@ app.UpgradeKeeper.SetUpgradeHandler(
...
+
+![AuthModuleDecisionTree](../assets/auth-module-decision-tree.png)
+
+**My authentication module needs to access IBC packet callbacks**
+
+Application developers that wish to consume IBC packet callbacks and react upon packet acknowledgements **must** continue using the controller submodule's legacy APIs. The authentication modules will not be need a `ScopedKeeper` anymore, though, because the channel capability will be claimed by the controller submodule. For example, given an Interchain Accounts authentication module keeper `ICAAuthKeeper`, the authentication module's `ScopedKeeper` (`scopedICAAuthKeeper`) is not needed anymore and can be removed for the argument list of the keeper constructor function, as shown here:
```
Comment (tmsdkeys) [inline docs/migrations/v5-to-v6.md:109]:
```suggestion
Application developers that wish to consume IBC packet callbacks and react upon packet acknowledgements **must** continue using the controller submodule's legacy APIs. The authentication modules will not need a `ScopedKeeper` anymore, though, because the channel capability will be claimed by the controller submodule. For example, given an Interchain Accounts authentication module keeper `ICAAuthKeeper`, the authentication module's `ScopedKeeper` (`scopedICAAuthKeeper`) is not needed anymore and can be removed for the argument list of the keeper constructor function, as shown here:
```

File: docs/migrations/v5-to-v6.md [hunk 11]
```diff
@@ -87,21 +87,86 @@ app.UpgradeKeeper.SetUpgradeHandler(
...
++ }
+```
+
+where `controllertypes` is an import alias for `"github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/controller/types"`.
+
+In addition, in this use case the authentication module does not need to implement the `IBCModule` interface anymore.
+
+**I do not need a custom authentication module anymore**
+
+If your authentication module does not have any extra functionality compared to the default authentication module added in ibc-go v6 (the `MsgServer`), or if you can use a generic authentication module, such as the `x/auth`, `x/gov` or `x/group` modules from the Cosmos SDK (for at least v0.46), then you can remove your authentication module completely and use instead the gRPC endpoints of the `MsgServer` or the CLI added in ibc-go v6.
```
Comment (tmsdkeys) [inline docs/migrations/v5-to-v6.md:167]:
```suggestion
If your authentication module does not have any extra functionality compared to the default authentication module added in ibc-go v6 (the `MsgServer`), or if you can use a generic authentication module, such as the `x/auth`, `x/gov` or `x/group` modules from the Cosmos SDK (v0.46 and later), then you can remove your authentication module completely and use instead the gRPC endpoints of the `MsgServer` or the CLI added in ibc-go v6.
```

File: docs/apps/interchain-accounts/legacy/keeper-api.md [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+order: 3
+-->
+
+# Keeper API
+
+## Deprecation Notice
+
+**This document is deprecated and will be removed in future releases**.
+
+The controller submodule keeper exposes legacy two functions that allow custom authentication modules to register interchain accounts and send packets to the interchain account.
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/legacy/keeper-api.md:11]:
```suggestion
The controller submodule keeper exposes two legacy functions that allow respectively for custom authentication modules to register interchain accounts and send packets to the interchain account.
```

File: docs/apps/interchain-accounts/legacy/keeper-api.md [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+timeoutTimestamp := obtainTimeoutTimestamp()
+
+// Send the interchain accounts packet, returning the packet sequence
+// A nil channel capability can be passed, since the controller submodule (and not the authentication module) 
+// claims the channel capability since ibc-go v6.
+seq, err = keeper.icaControllerKeeper.SendTx(ctx, nil, portID, packetData, timeoutTimestamp)
+```
+
+The data within an `InterchainAccountPacketData` must be serialized using a format supported by the host chain. 
+If the host chain is using the ibc-go host chain submodule, `SerializeCosmosTx` should be used. If the `InterchainAccountPacketData.Data` is serialized using a format not support by the host chain, the packet will not be successfully received.
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/legacy/keeper-api.md:121]:
```suggestion
If the host chain is using the ibc-go host chain submodule, `SerializeCosmosTx` should be used. If the `InterchainAccountPacketData.Data` is serialized using a format not supported by the host chain, the packet will not be successfully received.  
```

File: docs/apps/interchain-accounts/legacy/keeper-api.md [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+The data within an `InterchainAccountPacketData` must be serialized using a format supported by the host chain. 
+If the host chain is using the ibc-go host chain submodule, `SerializeCosmosTx` should be used. If the `InterchainAccountPacketData.Data` is serialized using a format not support by the host chain, the packet will not be successfully received.  
+
+### `OnAcknowledgementPacket`
+
+Controller chains will be able to access the acknowledgement written into the host chain state once a relayer relays the acknowledgement. 
+The acknowledgement bytes will be passed to the auth module via the `OnAcknowledgementPacket` callback. 
+Auth modules are expected to know how to decode the acknowledgement. 
+
+If the controller chain is connected to a host chain using the host module on ibc-go, it may interpret the acknowledgement bytes as follows:
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/legacy/keeper-api.md:129]: Maybe you can link to the code where we can find the content of the ack, [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/host/keeper/relay.go#L31) if I'm correct?

File: docs/apps/interchain-accounts/legacy/integration.md [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+
+Chains who wish to support ICS-27 may elect to act as a host chain, a controller chain or both. Disabling host or controller functionality may be done statically by excluding the host or controller module entirely from the `app.go` file or it may be done dynamically by taking advantage of the on-chain parameters which enable or disable the host or controller submodules. 
+
+Interchain Account authentication modules are the base application of a middleware stack. The controller submodule is the middleware in this stack.
+
+Please note that since ibc-go v6 the channel capability is claimed by the controller submodule and therefore it is not required for authentication modules to claim the capability in the `OnChanOpenInit` callback. Therefore the custom authentication module does not need a scoped keeper anymore.
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/legacy/integration.md:19]: Perhaps add `>`for additional formatting to highlight this is a note?

File: docs/apps/interchain-accounts/auth-modules.md [hunk 16]
```diff
@@ -1,399 +1,21 @@
...
+order: 3
 -->
 
 # Building an authentication module
 
-### Deprecation Notice
+Authentication modules enable application developers to perform custom logic when interacting with the Interchain Accounts controller sumbmodule's `MsgServer`. {synopsis}
 
-**This document is deprecated and will be removed in future releases**.
+The controller submodule is used for account registration and packet sending. It executes only logic required of all controllers of interchain accounts. The type of authentication used to manage the interchain accounts remains unspecified. There may exist many different types of authentication which are desirable for different use cases. Thus the purpose of the authentication module is to wrap the controller submodule with custom authentication logic.
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/auth-modules.md:9]: Some diagrams might be helpful? Adi has posted quite a few in [this blog post](https://medium.com/the-interchain-foundation/ibc-go-v6-changes-to-interchain-accounts-and-how-it-impacts-your-chain-806c185300d7)

File: docs/apps/interchain-accounts/legacy/auth-modules.md [hunk 17]
```diff
@@ -0,0 +1,381 @@
...
+
+The controller submodule is used for account registration and packet sending. It executes only logic required of all controllers of interchain accounts. The type of authentication used to manage the interchain accounts remains unspecified. There may exist many different types of authentication which are desirable for different use cases. Thus the purpose of the authentication module is to wrap the controller submodule with custom authentication logic.
+
+In ibc-go, authentication modules are connected to the controller chain via a middleware stack. The controller submodule is implemented as [middleware](https://github.com/cosmos/ibc/tree/master/spec/app/ics-030-middleware) and the authentication module is connected to the controller submodule as the base application of the middleware stack. To implement an authentication module, the `IBCModule` interface must be fulfilled. By implementing the controller submodule as middleware, any amount of authentication modules can be created and connected to the controller submodule without writing redundant code.
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/legacy/auth-modules.md:15]: Add some diagrams? Same comment as in main auth-modules readme

File: docs/apps/interchain-accounts/legacy/auth-modules.md [hunk 18]
```diff
@@ -0,0 +1,381 @@
...
+    packet channeltypes.Packet,
+    relayer sdk.AccAddress,
+) error {
+    // perform custom logic
+
+    return nil
+}
+```
+
+The following functions must be defined to fulfill the `IBCModule` interface, but they will never be called by the controller submodule so they may error or panic.
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/legacy/auth-modules.md:95]: I think it's always useful to remind people that ICA is asymmetrical, which explains why the below call backs should panic

File: docs/apps/interchain-accounts/legacy/auth-modules.md [hunk 19]
```diff
@@ -0,0 +1,381 @@
...
+The data within an `InterchainAccountPacketData` must be serialized using a format supported by the host chain. 
+If the host chain is using the ibc-go host chain submodule, `SerializeCosmosTx` should be used. If the `InterchainAccountPacketData.Data` is serialized using a format not support by the host chain, the packet will not be successfully received.  
+
+### `OnAcknowledgementPacket`
+
+Controller chains will be able to access the acknowledgement written into the host chain state once a relayer relays the acknowledgement. 
+The acknowledgement bytes will be passed to the auth module via the `OnAcknowledgementPacket` callback. 
+Auth modules are expected to know how to decode the acknowledgement. 
+
+If the controller chain is connected to a host chain using the host module on ibc-go, it may interpret the acknowledgement bytes as follows:
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/legacy/auth-modules.md:260]: similar comments as in apps/interchain-accounts/auth-modules.md

File: docs/apps/interchain-accounts/development.md [hunk 20]
```diff
@@ -1,36 +1,36 @@
...
+To faciliate such an action, the controller submodule may be provided an underlying application which redirects to smart contract callers. 
 An improved design has been suggested in [ADR 008](https://github.com/cosmos/ibc-go/pull/1976) which performs this action via middleware. 
 
 Implementors of this use case are recommended to follow the ADR 008 approach.
 The underlying application may continue to be used as a short term solution for ADR 008 and the [legacy API](./auth-modules.md#registerinterchainaccount) should continue to be utilized in such situations. 
 
 ## Packet callbacks
 
-If a developer requires access to packet callbacks for their use case, then they having the following options:
+If a developer requires access to packet callbacks for their use case, then they have the following options:
```
Comment (tmsdkeys) [inline docs/apps/interchain-accounts/development.md:29]: What if a dev requires access to packet callbacks, but does not have Cosmwasm or any other smart contract capability enabled on the chain? What happens in this case when the legacy api is no longer available?

**Review by tmsdkeys — COMMENTED** [review]
Thanks  @crodriguezvega for restructuring the docs on ICA into main post v6 and legacy folders. This makes a lot of sense to me and will make it easier to remove the folder when the legacy API is removed.

Left a few comments, mostly nits/typos

File: docs/apps/interchain-accounts/development.md (hunk 20 above)
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/development.md:29]: I believe we will not deprecate the legacy API until that use case is covered by using the `MsgServer` + ADR-008.

File: docs/apps/interchain-accounts/active-channels.md (hunk 6 above)
Comment (colin-axner) [reply docs/apps/interchain-accounts/active-channels.md]: `NewMsgRegisterInterchainAccount` did not exist, so I think the recommended approach was `NewMsgChannelOpenInit` as it could be done without going through an on chain application

File: docs/apps/interchain-accounts/integration.md [hunk 21]
```diff
@@ -73,40 +72,35 @@ scopedICAAuthKeeper := app.CapabilityKeeper.ScopeToModule(icaauthtypes.ModuleNam
...
 
 // ICA auth AppModule
 icaAuthModule := icaauth.NewAppModule(appCodec, app.ICAAuthKeeper)
 
-// ICA auth IBC Module
-icaAuthIBCModule := icaauth.NewIBCModule(app.ICAAuthKeeper)
-
 // Create controller IBC application stack and host IBC module as desired
-icaControllerStack := icacontroller.NewIBCMiddleware(icaAuthIBCModule, app.ICAControllerKeeper)
+icaControllerStack := icacontroller.NewIBCMiddleware(nil, app.ICAControllerKeeper)
```
Comment (colin-axner) [inline docs/apps/interchain-accounts/integration.md:97]: Hmm, should we be creating a stack with the auth module?

File: docs/apps/interchain-accounts/legacy/integration.md [hunk 22]
```diff
@@ -0,0 +1,193 @@
+<!--
+order: 1
+-->
+
+# Integration
```
Comment (colin-axner) [inline docs/apps/interchain-accounts/legacy/integration.md:5]: were modifications made to the legacy files that need to be reviewed?

File: docs/apps/interchain-accounts/active-channels.md (hunk 6 above)
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/active-channels.md]: >NewMsgRegisterInterchainAccount did not exist

True, I was thinking about the message to register interchain accounts that existed in the inter-tx app... Anyway, I reverted the change.

File: docs/apps/interchain-accounts/legacy/auth-modules.md (hunk 19 above)
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/legacy/auth-modules.md:148]: I forgot to remove it from one of the places! :)

File: docs/apps/interchain-accounts/legacy/integration.md (hunk 22 above)
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/legacy/integration.md:5]: I think the only changes I made were in the first paragraphs of each page, maybe to clarify the legacy idea a bit more. But the rest of the content (pseudocode, etc) I think it's the same as before.

File: docs/apps/interchain-accounts/active-channels.md [hunk 23]
```diff
@@ -1,27 +1,31 @@
...
+}
 ```
 
 Alternatively, any relayer operator may initiate a new channel handshake for this interchain account once the previously set `Active Channel` is in a `CLOSED` state. This is done by initiating the channel handshake on the controller chain using the same portID associated with the interchain account in question.  
 
-It is important to note that once a channel has been opened for a given Interchain Account, new channels can not be opened for this account until the currently set `Active Channel` is set to `CLOSED`. 
+It is important to note that once a channel has been opened for a given interchain account, new channels can not be opened for this account until the currently set `Active Channel` is set to `CLOSED`. 
 
-## Future Improvements
+## Future improvements
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/active-channels.md:28]: Yes, it's intentional because Active Channels is uppercase since it is like an official name.

File: docs/apps/interchain-accounts/active-channels.md [hunk 24]
```diff
@@ -1,27 +1,31 @@
...
 
 In the case of a channel closing, a controller chain needs to be able to regain access to the interchain account registered on this channel. `Active Channels` enable this functionality. 
-When an Interchain Account is registered using `MsgRegisterInterchainAccount`, a new channel is created on a particular port. During the `OnChanOpenAck` and `OnChanOpenConfirm` steps (controller & host chain) the `Active Channel` for this interchain account
-is stored in state.
+
+When an Interchain Account is registered using `MsgRegisterInterchainAccount`, a new channel is created on a particular port. During the `OnChanOpenAck` and `OnChanOpenConfirm` steps (on controller & host chain respectively) the `Active Channel` for this interchain account is stored in state.
 
 It is possible to create a new channel using the same controller chain portID if the previously set `Active Channel` is now in a `CLOSED` state. This channel creation can be initialized programatically by sending a new `MsgChannelOpenInit` message like so:
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/active-channels.md:13]: You mean the generated interchain accounts address?

File: docs/apps/interchain-accounts/development.md [hunk 25]
```diff
@@ -1,36 +1,36 @@
...
 # Development use cases
 
-The initial version of interchain accounts allowed for the controller module to be extended by providing it with an underlying application which would handle all packet callbacks.
+The initial version of Interchain Accounts allowed for the controller submodule to be extended by providing it with an underlying application which would handle all packet callbacks.
 That functionality is now being deprecated in favor of alternative approaches. 
 This document will outline potential use cases and redirect each use case to the appropriate documentation. 
 
 ## Custom authentication 
 
 Interchain accounts may be associated with alternative types of authentication relative to the traditional public/private key signing.
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/development.md:13]: I will leave as is, since here we refer to the account, not the feature (I make that distinction when using uppercase or not).

File: docs/apps/interchain-accounts/development.md [hunk 26]
```diff
@@ -1,36 +1,36 @@
...
 Interchain accounts may be associated with alternative types of authentication relative to the traditional public/private key signing. 
-If you wish to develop or use interchain accounts with a custom authentication module, we recommend you use ibc-go v6 or greater. 
+If you wish to develop or use Interchain Accounts with a custom authentication module and do not need to execute custom logic on the packet callbacks, we recommend you use ibc-go v6 or greater and that your custom authentication module interacts with the controller submodule via the [`MsgServer`](./messages.md).
 
-The custom authentication module should interact with the controller module via the [MsgServer](./messages.md).
+If you wish to consume and execute custom logic in the packet callbacks, then please read the section [Packet callbacks](#packet-callbacks) below.
 
 ## Redirection to a smart contract
 
 It may be desirable to allow smart contracts to control an interchain account.
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/development.md:20]: Same, I will leave it lowercase because we refer to the account not the feature. I hope that makes sense!

File: docs/apps/interchain-accounts/legacy/keeper-api.md (hunk 14 above)
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/legacy/keeper-api.md]: Instead of linking to code, I added a clarification of what the acknowledgement contains. I hope that's ok.

File: docs/apps/interchain-accounts/legacy/auth-modules.md [hunk 27]
```diff
@@ -0,0 +1,381 @@
...
+
+The authentication module must:
+
+- Authenticate interchain account owners.
+- Track the associated interchain account address for an owner.
+- Send packets on behalf of an owner (after authentication).
+
+Please note that since ibc-go v6 the channel capability is claimed by the controller submodule and therefore it is not required for authentication modules to claim the capability in the `OnChanOpenInit` callback. When the authentication module sends packets on the channel created for the associated interchain account it can pass a `nil` capability to the legacy function `SendTx` of the controller keeper (see [section `SendTx`](#sendtx) below for mode information). 
+
+## `IBCModule` implementation
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/apps/interchain-accounts/legacy/auth-modules.md:25]: That's a good observation! However, I think I prefer to list all the callbacks, just to be very explicit.

**Comment by crodriguezvega** [comment]
Splitted in #2891, #2892 and #2893
